src/index.ts

```diff
@@ -81,8 +81,11 @@ function filterInternal<T, R extends Runtype<T>>(
 
 			return filterInternal(<any>alt, x, context);
 		case "constraint":
-		case "optional":
 			return filterInternal(<any>r.underlying, x, context);
+		case "optional":
+			return typeof x === "undefined"
```

Comment:
Does this need a `null` check as well?

Reply:
Just pushed some tests in 3299817. The Runtypes `optional` type only supports `undefined`, not `null`.